client/src/partials/exchange_rate/exchange_rate.js

```diff
@@ -1,114 +1,125 @@
 angular.module('bhima.controllers')
-.controller('exchangeRate', [
-  '$scope',
-  'connect',
-  'messenger',
-  'appstate',
-  'validate',
-  'exchange',
-  'util',
-  function ($scope, connect, messenger, appstate, validate, exchange, util) {
-    var dependencies = {};
-    var session = $scope.session = {};
-
-    dependencies.currency = {
-      required : true,
-      query : {
-        tables : {
-          'currency' : { 'columns' : ['id', 'name', 'symbol', 'note']
-          }
+.controller('ExchangeRateController', ExchangeRateController);
+
+ExchangeRateController.$inject = [
+  'connect', 'appstate', 'validate', 'exchange', 'SessionService', 'DateService'
+];
+
+/**
+* This controller works in tandem with the ExchangeRateService to allow a user to
+* set an exchange rate for a given day.
+*
+* TODO
+*   - do not require a new exchange rate every day (see issue #619).
+*   - rename service `exchange` to be `ExchangeRateService`
+*
+* @controller ExchangeRateController
+*/
+function ExchangeRateController(connect, appstate, validate, exchange, Session, Dates) {
+  var vm = this;
+  var dependencies = {};
+
+  // bind data
+  vm.today = Dates.current.day();
+  vm.enterprise = Session.enterprise;
+  vm.form = { date : vm.today };
+  vm.state = 'default';
+
+  vm.submit    = submit;
+  vm.fcurrency = fcurrency;
+  vm.setState  = setState;
+
+  /* ------------------------------------------------------------------------ */
+
+  dependencies.currency = {
+    required : true,
+    query : {
+      tables : {
+        'currency' : { 'columns' : ['id', 'name', 'symbol', 'note']
         }
       }
-    };
+    }
+  };
 
-    dependencies.rates = {
-      query : {
-        tables : {
-          'exchange_rate' : {
-            'columns' : ['id', 'enterprise_currency_id', 'foreign_currency_id', 'rate', 'date']
-          }
+  dependencies.rates = {
+    query : {
+      tables : {
+        'exchange_rate' : {
+          'columns' : ['id', 'enterprise_currency_id', 'foreign_currency_id', 'rate', 'date']
         }
       }
-    };
+    }
+  };
 
-    $scope.maxDate = util.htmlDate(new Date());
-    $scope.currentDate = util.htmlDate(new Date());
-    session.is_oldDate = false;
-    session.oldDate = new Date();
+  //
```

Comment:
Your suggestion is essentially #709.  This PR is only a bug fix to address the bugs in #588 and #673.